### PR TITLE
[SDPA-6303] Fixes an issue that causes link field broken

### PR DESCRIPTION
**Pull request by vincent-gao**
### Jira
https://digital-engagement.atlassian.net/browse/SDPA-6303

### Issue
see the ticket.

### CODE REVIEW

**Review by mayurngondhkar — APPROVED** [review]
LGTM

File: src/EntityAutocompleteMatcher.php [hunk 1]
```diff
@@ -27,6 +28,12 @@ public function getMatches($target_type, $selection_handler, $selection_settings
 
     $handler = $this->selectionManager->getInstance($options);
 
+    // If the handler cannot be found, returns the default one.
+    if ($handler instanceof Broken) {
```
Comment (christopher-hopper) [inline src/EntityAutocompleteMatcher.php:32]: @vincent-gao 
Just wondering under what circumstances is Drupal not able to load the actual EntityReferenceSelection plugin? Are we covering up some other config or code problem? Which plugin was it trying to load when it returned "Broken" instead?

Comment (christopher-hopper) [reply src/EntityAutocompleteMatcher.php:32]: I read the bug report ticket but couldn't see any clues.

Comment (christopher-hopper) [reply src/EntityAutocompleteMatcher.php:32]: @vincent-gao 
I think I see why it cannot load the EntityReferenceSelection handler. There's no hander called "tide_core". 

It should be looking for "tide:node" not "tide_core" at line 20.

The handler is defined here:
https://github.com/dpc-sdp/tide_core/blob/develop/src/Plugin/EntityReferenceSelection/TideNodeSelection.php 

```php
/**
 * Provides labels for the node entity type.
 *
 * @EntityReferenceSelection(
 *   id = "tide:node",
 *   label = @Translation("Tide node selection"),
 *   entity_types = {"node"},
 *   group = "tide",
 *   weight = 1,
 *   base_plugin_label = "Tide"
 * )
 */
```

If you fix that, it should no longer return the "broken" handler as a fallback I think. Can you try that out?

**Review by christopher-hopper — CHANGES_REQUESTED** [review]

**Comment by christopher-hopper** [comment]
@vincent-gao 
I think this might be the fix:
https://github.com/dpc-sdp/tide_core/pull/326/files 

Haven't tested it, but I think this is what's going on.

File: src/EntityAutocompleteMatcher.php (hunk 1 above)
Comment (vincent-gao) [reply src/EntityAutocompleteMatcher.php]: Thanks @christopher-hopper ! I think it works.

**Review by christopher-hopper — APPROVED** [review]

**Pull request merged**